[F10-3] Heart²

liaujianjie opened the pull request:
![45598280-f0abd600-ba0b-11e8-8202-2c607cf221ea](https://user-images.githubusercontent.com/3143132/45919868-e471be00-bece-11e8-93af-395b225b0daf.png)

-----

@eehooi @wailunlim @NightYeti @dongsiji @liaujianjie

dalessr commented on README.adoc:
```diff
@@ -1,7 +1,7 @@
 = Address Book (Level 4)
```
Can change it to the name of your app. Remember to change all `Address Book Level 4` in your project

dalessr commented on README.adoc:
```diff
@@ -1,7 +1,7 @@
-https://travis-ci.org/se-edu/addressbook-level4[image:https://travis-ci.org/se-edu/addressbook-level4.svg?branch=master[Build Status]]
+https://travis-ci.org/CS2103-AY1819S1-F10-3/main[image:https://travis-ci.org/CS2103-AY1819S1-F10-3/main.svg?branch=master[Build Status]]
 https://ci.appveyor.com/project/damithc/addressbook-level4[image:https://ci.appveyor.com/api/projects/status/3boko2x2vr5cc3w2?svg=true[Build status]]
 https://coveralls.io/github/se-edu/addressbook-level4?branch=master[image:https://coveralls.io/repos/github/se-edu/addressbook-level4/badge.svg?branch=master[Coverage Status]]
```
Can change all these links to your own project's build. (For Appveyor can use team lead's project link)

dalessr commented on docs/AboutUs.adoc:
```diff
@@ -4,53 +4,48 @@
 
-AddressBook - Level 4 was developed by the https://se-edu.github.io/docs/Team.html[se-edu] team. +
-_{The dummy content given below serves as a placeholder to be used by future forks of the project.}_ +
+Heart² was developed by the F10-3 team. +
```
Is it better to specify as `NUS School of Computing AY1819...`?

dalessr commented on docs/AboutUs.adoc:
```diff
@@ -4,53 +4,48 @@
-{empty}[http://www.comp.nus.edu.sg/~damithch[homepage]] [https://github.com/damithc[github]] [<<johndoe#, portfolio>>]
+=== Dong SiJi
+image::dongsiji.jpg[width="150", align="left"]
+{empty}[https://github.com/dongsiji[github]] [<<johndoe#, portfolio>>]
```
Why here `johndoe#` still appears?

dalessr commented on docs/AboutUs.adoc:
```diff
@@ -4,53 +4,48 @@
+{empty}[https://github.com/dongsiji[github]] [<<johndoe#, portfolio>>]
 
-Role: Project Advisor
+Role: In charge of ...
```
Can decide the role now. Feel free to modify if there are changes during the process.

dalessr reviewed (commented):
Nice work! Some suggestions below.

Also, please update user guide (i.e. features for each version) and developer guide (i.e. user story, use cases, NFR...) accordingly.

dalessr reviewed (commented):
Good job 👍 @eehooi @wailunlim @NightYeti @dongsiji @liaujianjie

Some comments:
- AboutUs page: assign roles and components to specific team members (i.e. role: team lead/developer, component: storage/model/logic...)
- User Guide: add description and examples for all the features due to consistency. Can omit examples of trivial features (i.e. exit). Similar to the other features, features for v2.0 should also have description, format & examples.
- Developer Guide: some use cases are still incorrect. Remember that the steps in MSS reflect the interaction between user and system, which indicates that 2 system actions cannot appear consecutively.  Can remove Appendix G if not needed.

dalessr reviewed (commented):
@eehooi @wailunlim @NightYeti @dongsiji @liaujianjie Some comments:

Contact Us:
- Link to the issue tracker should be the link of your team repo's issue tracker instead of se-edu

User Guide:
- For consistency, add examples for every non-trivial commands (i.e. login)
- Add description, format and example for 3.13 also, even though you will not be implementing it.

Developer Guide:
- Update/add diagrams where applicable. Some class/sequence diagrams do not follow your current product any more. i.e.:
![image1](https://user-images.githubusercontent.com/22449685/47213374-615c5e80-d3cd-11e8-9378-06bd2bf9e6c6.png)

![image2](https://user-images.githubusercontent.com/22449685/47213380-65887c00-d3cd-11e8-9e6e-a437b64ae23f.png)

- Add labels for all the newly added diagrams

dongsiji committed:
Add introduction

dongsiji committed:
Add about this document

dongsiji committed:
Add ug and dg into pp

dongsiji committed:
Fix introduction and about this document

dongsiji committed:
Complete writing the pp

dongsiji committed:
Add other contributions

liaujianjie committed:
Add some content to project portfolio

liaujianjie committed:
Update overview

liaujianjie committed:
Add more portfolio content

dongsiji committed:
Add bold

Commit:
Update portfolio

Added work on command parser

Commit:
Update portfolio

Stopped halfway at the first major enhancement: extending commands

dongsiji committed:
Update delete, select autocommand parser

liaujianjie committed:
Merge pull request #279 from liaujianjie/undo-redo-typo

Replace quotes with backticks for undo and redo in UG

dongsiji committed:
Remove unused imports

liaujianjie committed:
Small fixes and changes to UG

liaujianjie committed:
Add new icon images for notes, tips and danger

liaujianjie committed:
Add tests for AutoMatchCommandParser

liaujianjie committed:
Add tests for AutoMatchCommand constructor

Commit:
Change name

Commit:
Update strings to use word "view" instead of "select"

liaujianjie committed:
Add tests for ContactType#fromString

Commit:
Minor update to strings related to view command

wailu committed:
Merge pull request #292 from wailunlim/update-name-ppp

Change name

liaujianjie committed:
Add tests for AutoMatchCommand#equals

Commit:
Minor change to string in test

liaujianjie committed:
Refactor AutoMatchCommand to use integer for id

Commit:
Update Add feature

dongsiji committed:
Merge pull request #287 from dongsiji/update-parser

Update delete, select autocommand parser

dongsiji committed:
Merge pull request #288 from dongsiji/update-ug

Specify ID to be an integer

wailu committed:
Merge pull request #293 from wailunlim/change-select-command-word-to-view

Change select command word to view

liaujianjie committed:
Merge pull request #289 from liaujianjie/ug-fixes

Small fixes and changes to UG

liaujianjie committed:
Merge branch 'master' into automatch-tests

liaujianjie committed:
Merge pull request #290 from liaujianjie/guide-icons

Add icons for tips, notes and danger

Commit:
Update remove clients and vendors feature

liaujianjie committed:
Merge pull request #291 from liaujianjie/automatch-tests

Add tests for auto-matching

eehooi committed:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-F10-3/main

ganchinyao committed:
change all instances of service providers to vendors in UG and DG

ganchinyao committed:
Remove unused Appendix on product survey in DG

ganchinyao committed:
Change instances of service providers to vendors in PPP

liaujianjie committed:
Add more changes to portfolio

liaujianjie committed:
Merge pull request #226 from CS2103-AY1819S1-F10-3/jianjie-portfolio

Add some content to project portfolio

ganchinyao committed:
Minor english edit

eehooi committed:
Update logo

ganchinyao committed:
Show the user role at the bottom center of screen

ganchinyao committed:
Change storage xml file name

ganchinyao committed:
Merge branch 'master' into refactor-serviceproviders

ganchinyao committed:
change folder directory of ppp images

ganchinyao committed:
Merge pull request #297 from NightYeti/role-feedback

Show user's role at bottom center of screen

ganchinyao committed:
Merge pull request #298 from NightYeti/change-image-directory

change folder directory of ppp images

ganchinyao committed:
Merge pull request #294 from NightYeti/refactor-serviceproviders

Change instances of service providers to vendors

eehooi committed:
Merge pull request #296 from eehooi/change-app-logo

Change app logo

ganchinyao committed:
Remove some discrete tags in DG

ganchinyao committed:
Remove appendix on manual testing in DG

ganchinyao committed:
Make all figure description italics

ganchinyao committed:
Allow client and vendor to be the same person

dalessr reviewed (commented):
@eehooi @wailunlim @NightYeti @dongsiji @liaujianjie Some comments:

In Contact Us: the link for `issue tracker` should be the link of your team repo's issue tracker instead of `SE-EDU`.

In Developer Guide: 
![default](https://user-images.githubusercontent.com/22449685/48295643-e1577f00-e4c8-11e8-82dd-5174059ef5ea.png)
Here is it trying to call the method inside the class? If true, should it be drawn like that?

ganchinyao committed:
Change instances of address book in the application to heart2

ganchinyao committed:
Update issue link in contactus, minor english change in docs

ganchinyao committed:
Minor update to UG

ganchinyao committed:
Add stylesheet to PPP

ganchinyao committed:
Minor update to PPP and UG

Commit:
Add no write privilege add test

Commit:
Add no delete privilege delete test

eehooi committed:
Create default data

eehooi committed:
Fix checkstyle error

ganchinyao committed:
Remove unused imports

eehooi committed:
Merge pull request #301 from eehooi/create-default-data

Create default data

ganchinyao committed:
Merge branch 'master' into enhancements

ganchinyao committed:
update ui.png

ganchinyao committed:
Update Ui.png

Commit:
Add invalid args test for ListCommandParser

Commit:
Add no write privilege update test

Commit:
Add required identifier but missing tests && update buggy update tests

Update a small comment

Commit:
Add parsing commands before logging in tests

Commit:
Add test for parsing unnecessary identifier in commands

Commit:
Split getAddCommand for client and vendor

Commit:
Add vendor tests for parser

Commit:
Add required non empty arguments but present tests

wailu committed:
Merge pull request #300 from wailunlim/add-tests-for-uncovered

Add tests for uncovered lines

ganchinyao committed:
Merge branch 'master' into enhancements

Commit:
Fix wrong string feedback for update and addService

Add their test cases too

ganchinyao committed:
Remove a test case since it is now possible to have both vendor and client same person

Commit:
Fix illegal identifier not giving illegal command format

This is for register account and change password commands. Commit contains their test cases too.

ganchinyao committed:
Add line to fix eof checkstyle

ganchinyao committed:
Merge pull request #299 from NightYeti/enhancements

Update documentations, client/vendor same name bug fix

Commit:
Fix test cases strings

Commit:
Abide to checkstyle

wailu committed:
Merge pull request #303 from wailunlim/fix-wrong-error-message-displayed

Fix wrong string feedback shown to user

ganchinyao committed:
Fix the bug of having duplicated contact after updating entries

ganchinyao committed:
Update UG to reflect the requiremenet of adding and updating

ganchinyao committed:
Update UG to reflect the conditions of some commands

ganchinyao committed:
Fix typo

Commit:
Change limwailun -> wailunlim

ganchinyao committed:
Merge pull request #307 from NightYeti/bug-updatecommand

Fix a bug in update command causing entire data to disappear

wailu committed:
Merge pull request #308 from wailunlim/fix-portfolio-not-properly-linked-to-page

Change limwailun -> wailunlim

ganchinyao committed:
Update dg on manual testing

Commit:
Add unique ID feature into DG

Commit:
Edit delete contact feature portion in DG

ganchinyao committed:
Merge pull request #309 from NightYeti/dg-manualtest

Update dg on manual testing

Commit:
Edit update contact portion in DG

Commit:
Small change in phrasing

Commit:
Add design considerations for update feature in DG

Commit:
Add a design consideration for add feature in DG

Commit:
Add markup for "Aspect"

Commit:
Add whitespace

dongsiji committed:
Update format

dongsiji committed:
Update callouts

dongsiji committed:
Update login

dongsiji committed:
Update example

dongsiji committed:
Update login, change password

dongsiji committed:
Update list all the way to update

dongsiji committed:
Update addservice

dongsiji committed:
Update command summary

Commit:
Rephrase sentence in update feature in DG

dongsiji committed:
Update tag

Commit:
Delete redundant section in delete feature and add view feature

Commit:
Add and update sequence diagram for delete command

Commit:
Change location of image

Previously in wrong folder

Commit:
Delete unused (replaced) diagram

Commit:
Update delete diagram

Commit:
Update diagram

Commit:
Update references to old command syntax

dongsiji committed:
Standardise formatting

eehooi committed:
Update parts of UG and DG

dongsiji committed:
Update command summary

eehooi committed:
Update Heart2 format

Commit:
Update and replace old diagrams

dongsiji committed:
Merge pull request #310 from dongsiji/update-ug

Update ug

Commit:
Add a tip that points to a diagram earlier in doc

eehooi committed:
Merge branch 'master' into update-ug-and-dg

dongsiji committed:
Update list command

eehooi committed:
Merge pull request #311 from eehooi/update-ug-and-dg

Update parts of UG and DG

Commit:
Update language && add undo tip

wailu committed:
Merge pull request #312 from wailunlim/update-ug-dg

Update DG

Commit:
Add "can redo" to tip

Commit:
Add a note for view

Regarding browser panel showing

Commit:
Update wording for addservice

Commit:
Add additional tip for view command

dongsiji committed:
Add sequence diagrams to list

dongsiji committed:
Add manual testing for list

dongsiji committed:
Update list in userguide

Commit:
Add enhancement on support for client and vendors

dongsiji committed:
Merge pull request #314 from dongsiji/update-ugdg

Update ugdg

eehooi committed:
Update DG appendix

eehooi committed:
Merge branch 'master' into update-appendix-dg

eehooi committed:
Update Use Case

eehooi committed:
Merge branch 'update-appendix-dg' of https://github.com/eehooi/main into update-appendix-dg

eehooi committed:
Change title

Commit:
Add major enhancement on unique ID feature

eehooi committed:
Merge pull request #315 from eehooi/update-appendix-dg

Update DG appendix

wailu committed:
Merge pull request #313 from wailunlim/update-ug-dg

Update UG

Commit:
Add minor enhancement on parser

ganchinyao committed:
Update callouts image of UG to default callouts

Commit:
Add minor enhancement on extending commands

Commit:
Delete outdated texts

ganchinyao committed:
Minor edit

ganchinyao committed:
Minor edit to DG manual testing

ganchinyao committed:
Fix [Warning] callout not imported to UG

ganchinyao committed:
Make it clear in UG that undo, redo, and history are only for this session

Commit:
Add to code contributed section and other contributions

Commit:
Add elaboration for code contribution section

Commit:
Update formatting and make minor changes to language

wailu committed:
Merge pull request #318 from wailunlim/update-project-portfolio-wailun

Update project portfolio for myself

ganchinyao committed:
Merge pull request #316 from NightYeti/ug-callouts

Change current UG callouts to the default callouts

ganchinyao committed:
Remove all states in VersionedAddressBook after user log out

ganchinyao committed:
Write test case for the 2 new methods added

ganchinyao committed:
Update UG to reflect the history clearing upon logging out

ganchinyao committed:
Remove unused test cases

ganchinyao committed:
Remove unuse imports

Commit:
Add numberings for major and minor enhancements

Commit:
Spelling

ganchinyao committed:
Remove unuse test case

ganchinyao committed:
Update PPP to reflect changes in UG

eehooi committed:
Update AddService Implementation for UG and DG

ganchinyao committed:
Merge pull request #319 from NightYeti/bug-undo

Fix the bug where logging out and logging in can undo states

ganchinyao committed:
Merge pull request #320 from NightYeti/remove-skipped-test

Remove unused test cases

eehooi committed:
Update PPP

ganchinyao committed:
Merge pull request #322 from NightYeti/ppp-spelling

Update PPP to reflect changes in UG

eehooi committed:
Merge pull request #323 from eehooi/update-addservice-ug-dg

Update AddService Implementation for UG and DG

eehooi committed:
Merge pull request #324 from eehooi/update-project-portfolio

Update PPP

Commit:
Edit note about duplicates

Commit:
Add note and a tip on editing tags

Commit:
Update UG

wailu committed:
Merge pull request #325 from wailunlim/update-ug-dg

Update UG and PPP

Commit:
Add link to reposense for code contributed section in PPP

Commit:
Update contributions section

Commit:
Update code contribution

dongsiji committed:
Update ug and dg contributions

eehooi committed:
Fix minor formatting in DG and PPP

dongsiji committed:
Update language

eehooi committed:
Merge pull request #326 from eehooi/fix-minor-formatting-dg-ppp

Fix minor formatting in DG and PPP

dongsiji committed:
Update diagrams

eehooi committed:
Remove delete manual testing and fix minor PPP details

dongsiji committed:
Update summary of contributions

dongsiji committed:
Update link to functional code contributed

dongsiji committed:
Update other contributions

dongsiji committed:
Update contributed code for list

dongsiji committed:
Update dg contribution

dongsiji committed:
Update other contributions

dongsiji committed:
Update styling

Commit:
Update SD for delet

dongsiji committed:
Update command summary

dongsiji committed:
Add table for command summary

dongsiji committed:
Edit styling for table

dongsiji committed:
Change command summary to table format

dongsiji committed:
Add table to ppp

Commit:
Replace powerpoint file on SD

Commit:
Update the high level sequence diagram

dongsiji committed:
update ppp

dongsiji committed:
Update use cases

dongsiji committed:
Update command summary

dongsiji committed:
Add newline at EOF

dongsiji committed:
Change to use getType

dongsiji committed:
Change other contributions

ganchinyao committed:
Disallow sensitive commands from storing parameters to history

dongsiji committed:
Delete johndoe.adoc

dongsiji committed:
Add <ID> to command summary table

dongsiji committed:
Merge pull request #329 from dongsiji/update-collectionUtil

Change to use getType

dongsiji committed:
Merge pull request #328 from dongsiji/dongsiji-pp

Dongsiji pp

dongsiji committed:
Update diagrams

dongsiji committed:
Update to v1.4

Commit:
Add add client sequence diagram png and pptx

eehooi committed:
Merge pull request #327 from eehooi/update-ppp-and-delete-manual-testing

Remove delete manual testing and fix minor PPP details

dongsiji committed:
Merge pull request #331 from dongsiji/update-diagrams

Update diagrams

Commit:
Add explanation for add client sequence diagram

ganchinyao committed:
Merge pull request #330 from NightYeti/fix-history

Disallow sensitive commands from storing parameters to history

dongsiji committed:
Update tools

Commit:
Add UG DG segments to PPP

wailu committed:
Merge pull request #332 from wailunlim/update-ug-dg

Update DG

dongsiji committed:
Merge pull request #333 from dongsiji/update-ppp

Update tools

dongsiji committed:
Fix link for compiled code

dongsiji committed:
Add backticks to list

wailu committed:
Merge pull request #334 from wailunlim/update-project-portfolio-wailun

Finalise PPP

dongsiji committed:
Merge pull request #335 from dongsiji/update-ppp

Finalise ppp

Commit:
Correct minor spelling missteaks

Commit:
Correct some minor missteaks pt2

wailu committed:
Merge pull request #321 from wailunlim/correct-spelling

Correct spelling and some grammar

ganchinyao committed:
Update DG to tell user how to change default root account credentials

ganchinyao committed:
Minor change to PPP

Commit:
Add manual testing for add, delete, update,view

Commit:
Minor fixes to formatting and typos

wailu committed:
Merge pull request #337 from wailunlim/add-manual-testing

Add manual testing for add, delete, update,view

ganchinyao committed:
Merge pull request #336 from NightYeti/root-account

Update DG to tell user how to change default root account credentials

dongsiji committed:
Update undoredo sequence diagram

dongsiji committed:
Merge pull request #338 from dongsiji/update-diagram

Update undoredo sequence diagram

liaujianjie committed:
Update PPP

liaujianjie committed:
Append link to personal site in PPP

liaujianjie committed:
Add reposense link

liaujianjie committed:
Change reposense link

liaujianjie committed:
Merge pull request #339 from CS2103-AY1819S1-F10-3/finalise-jianjie-ppp

Update PPP
